src/libasr/codegen/llvm_utils.cpp

```diff
@@ -373,6 +373,129 @@ namespace LFortran {
         shift_end_point_by_one(list);
     }
 
+    llvm::Value* LLVMList::find_item_position(llvm::Value* list,
+                                    llvm::Value* item) {
+        llvm::Type* pos_type = std::get<2>(typecode2listtype["i64"]);
+        llvm::Value* current_end_point = LLVM::CreateLoad(*builder,
+                                        get_pointer_to_current_end_point(list));
+        llvm::AllocaInst *i = builder->CreateAlloca(pos_type, nullptr);
+        LLVM::CreateStore(*builder, llvm::ConstantInt::get(
+                                    context, llvm::APInt(32, 0)), i);
+        llvm::Value* tmp = nullptr;
+
+        /* Equivalent in C++:
+         * int i = 0;
+         * while(list[i] != item && end_point > i) {
+         *     i++;
+         * }
+         *
+         * if (i == end_point) {
+         *    std::cout << "The list does not contain the element";
+         * }
+         */
+
+        llvm::BasicBlock *loophead = llvm::BasicBlock::Create(context, "loop.head");
+        llvm::BasicBlock *loopbody = llvm::BasicBlock::Create(context, "loop.body");
+        llvm::BasicBlock *loopend = llvm::BasicBlock::Create(context, "loop.end");
+
+        // head
+        llvm_utils->start_new_block(loophead);
+        {
+            llvm::Value *cond = builder->CreateAnd(builder->CreateICmpNE(
+                read_item(list, LLVM::CreateLoad(*builder, i), false), item),
+                builder->CreateICmpSGT(current_end_point,
+                                       LLVM::CreateLoad(*builder, i)));
+            builder->CreateCondBr(cond, loopbody, loopend);
+        }
+
+        // body
+        llvm_utils->start_new_block(loopbody);
+        {
+            tmp = builder->CreateAdd(
+                        LLVM::CreateLoad(*builder, i),
+                        llvm::ConstantInt::get(context, llvm::APInt(32, 1)));
+            LLVM::CreateStore(*builder, tmp, i);
+        }
+        builder->CreateBr(loophead);
+
+        // end
+        llvm_utils->start_new_block(loopend);
+
+
+        llvm::Function *fn = builder->GetInsertBlock()->getParent();
+        llvm::BasicBlock *thenBB = llvm::BasicBlock::Create(context, "then", fn);
+        llvm::BasicBlock *elseBB = llvm::BasicBlock::Create(context, "else");
+        llvm::BasicBlock *mergeBB = llvm::BasicBlock::Create(context, "ifcont");
+
+        llvm::Value* cond = builder->CreateICmpEQ(
+                              LLVM::CreateLoad(*builder, i), current_end_point);
+        builder->CreateCondBr(cond, thenBB, elseBB);
+        builder->SetInsertPoint(thenBB);
+        {
+            LCompilersException("The list does not contain the element");
+            // Not working!
+        }
```

Comment:
I assume LCompilersException is used to stop the program. 
Here it `doesn't trigger` if the condition is `true`. 
So, I used `throw LCompilersException()` to resolve it, but this leads to `triggering` even when the condition is `false`.

Reply:
Well I feel `throw LCompilersException` will be executed always during compiling the code with LPython.

Reason - `LCompilersException("The list does not contain the element");` is a part of LPython's code so it will execute when LPython is called. Its not a part of generated LLVM code. What you want is generating LLVM code to throw exceptions.

Anyways, I will fix this in your branch and push the changes here.

Reply:
I see! Yup, thank you!